### PR TITLE
Fixes/article pdf services

### PR DESCRIPTION
**Pull request by Atharva-techstuff**

### CODE REVIEW

File: src/components/PDFViewer/PDFViewer.js [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
 import { Document, Page, pdfjs } from 'react-pdf';
-
 pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`;
 
 export const PDFViewer = ({ pdfLink }) => {
   const [numPages, setNumPages] = useState(null);
+  const [windowWidth, setWindowWidth] = useState(1.2);
+
+  useEffect(() => {
+    const width = window.innerWidth;
```
Comment (Madhavkabra) [inline src/components/PDFViewer/PDFViewer.js:10]: Needs to be improvised into separate function inside utils folder

File: src/components/Chips/Chips.js [hunk 2]
```diff
@@ -1,10 +1,28 @@
...
+
+  useEffect(() => {
+    router.push(`/articles?filter`, { query: selectedCategories });
+  }, [selectedCategories]);
+
+  const clickHandleCategory = ({ title, selectedCategories }) => {
+    if (title === 'Reset All') {
+      setSelectedCategories([]);
+    } else if (selectedCategories.includes(title)) {
+      selectedCategories.splice(selectedCategories.indexOf(title), 1);
```
Comment (Madhavkabra) [inline src/components/Chips/Chips.js:17]: Do not mutate the state directly

File: src/pages/articles/Articles.js [hunk 3]
```diff
@@ -151,16 +152,33 @@ const SkeletonPost = ({ index }) => {
 
 export const Articles = ({ posts, featured }) => {
   const [selectedCategories, setSelectedCategories] = useState([]);
+
+  const router = useRouter();
+  let urlPath = router.asPath;
+  const a1 = urlPath?.split('=');
```
Comment (Madhavkabra) [inline src/pages/articles/Articles.js:158]: Improvise naming

File: src/pages/articles/Articles.js [hunk 4]
```diff
@@ -190,23 +208,24 @@ export const Articles = ({ posts, featured }) => {
...
+            setSelectedCategories={setSelectedCategories}
           />
         ))}
       </div>
-
       {posts.map(({ slug, ...post }, index) => {
-        if (selectedCategories.every(elem => post.categories.includes(elem)))
+        if (
+          selectedCategories.some(elem => post.categories.includes(elem)) ||
+          selectedCategories.length == 0
```
Comment (Madhavkabra) [inline src/pages/articles/Articles.js:225]: No ==

**Review by Madhavkabra — CHANGES_REQUESTED** [review]

**Review by chandreshpatidar — APPROVED** [review]

**Pull request merged**